recipes/init.rb

```diff
@@ -54,3 +54,20 @@
     end
   end
 end
+
+include_recipe 'krb5' if hadoop_kerberos?
+
+krb5_principal "cdap/#{node['fqdn']}" do
```

Comment:
Backwards compatibility with previous CDAP versions requires we keep most of this stuff. I'll look at what I can remove. However, we shouldn't use `default['cdap']['kerberos']` as that's all the init script stuff. Rather, we should use the `cdap-site.xml` setting. This whole thing needs cleaning up, but that's outside the scope of this PR, which is just to switch to the LWRP. I'll clean up the LWRP invocations and file a JIRA for the rest.